### PR TITLE
split out verify_rangeproofs from sum_outputs

### PR DESCRIPTION
**Pull request by antiochp**
* split out verify_rangeproofs from sum_outputs
* added some timing to the debug msgs - 

```
Mar 20 14:00:25.036 DEBG Validated, summed (and offset) 4120 kernels, pmmr size 8237, took 4s
Mar 20 14:00:25.688 DEBG Summed 4117 Outputs, pmmr size 8245, took 0s
Mar 20 14:01:37.413 DEBG Verified 4117 Rangeproofs, pmmr size 8245, took 71s
```

Summing and offsetting the kernels is reasonably fast (4s for ~4,000 kernels).
Summing the outputs is fast.
Verifying the rangeproofs for each of these outputs is _slow_ (71s for ~4,000 rangeproofs).

### CODE REVIEW

File: chain/src/chain.rs [hunk 1]
```diff
@@ -415,6 +415,7 @@ impl Chain {
 	pub fn validate(&self) -> Result<(), Error> {
 		let header = self.store.head_header()?;
 		let mut txhashset = self.txhashset.write().unwrap();
+		txhashset.force_rollback();
```
Comment (antiochp) [inline chain/src/chain.rs]: This slipped in from another PR.

**Comment by ignopeverell** [comment]
We may have to look closer at batch verification for those range proofs.

**Comment by antiochp** [comment]
I wonder if its worth passing a `fast=true|false` flag into validate() to either validate all the rangeproofs, which we want to do for a fast sync vs. possibly skipping the rangeproof validation, say on check_compact, where we have presumably already validated all the rangeproofs but we still want to quickly re-validate the chain state?

**Comment by ignopeverell** [comment]
Seems like it would make sense.

**Pull request merged**